photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/GameTrainingDriver.scala

```diff
@@ -591,14 +609,29 @@ object GameTrainingDriver extends GameDriver {
    */
   private def calculateStatistics(
       data: DataFrame,
-      featureShards: Iterable[FeatureShardId]): FeatureShardStatistics =
+      featureShards: Iterable[FeatureShardId],
+      intercepts: Map[FeatureShardId, Option[Int]] = null)(implicit cm: ClassTag[Double]): FeatureShardStatistics =
     featureShards.map { featureShardId =>
       // Calling rdd explicitly here to avoid a typed encoder lookup in Spark 2.1
-      (featureShardId, BasicStatisticalSummary(
-        data.select(featureShardId)
-          .rdd
-          .map(_.getAs[SparkMLVector](0))))
-    }
+      val featuresAsBV: RDD[BreezeVector[Double]] = data.select(featureShardId)
+        .rdd
+        .map(x => mlToBreeze(x.getAs[SparkMLVector](0)))
+
+      val featureForSummary =
+        if(intercepts == null){
+          featuresAsBV
+        } else{
+          val featuresIndices: Set[Int] = featuresAsBV.first().keySet
+          val interceptIndex: Option[Int] = intercepts(featureShardId)
+          val featuresIndicesWithoutIntercept: Set[Int] = interceptIndex match {
+            case Some(intercept) => featuresIndices - intercept
+            case None => featuresIndices
+          }
+          featuresAsBV.map{v => v(featuresIndicesWithoutIntercept.toIndexedSeq).toVector}
+        }
```

Comment:
This intercept issue has become larger than we thought. As mentioned elsewhere, we should just hardcode the logic for now and return to it in a separte intercept refactor. I propose we use this block of code for now:
```
  private def calculateStatistics(
      data: DataFrame,
      featureIndexMapLoaders: Map[FeatureShardId, IndexMapLoader]): Map[FeatureShardId, BasicStatisticalSummary] =
    featureIndexMapLoaders
      .map { case (featureShardId, indexMapLoader) =>
        // Calling rdd explicitly here to avoid a typed encoder lookup in Spark 2.1
        val featuresAsBV: RDD[BreezeVector[Double]] = data
          .select(featureShardId)
          .rdd
          .map(x => VectorUtils.mlToBreeze(x.getAs[SparkMLVector](0)))

        val featuresForSummary = indexMapLoader
          .indexMapForDriver()
          .get(Constants.INTERCEPT_KEY)
          .map(_ => featuresAsBV.map(dropIntercept))
          .getOrElse(featuresAsBV)

        (featureShardId, BasicStatisticalSummary(featuresForSummary))
      }

  private def dropIntercept(baseVector: BreezeVector[Double]): BreezeVector[Double] = baseVector match {
    case dense: DenseVector[Double] =>
      new DenseVector[Double](dense.data, dense.offset, dense.stride, dense.length - 1)

    case sparse: SparseVector[Double] =>
      new SparseVector[Double](sparse.index.filterNot(_ < sparse.length - 1), sparse.data, sparse.length - 1)
  }
```

Reply:
what if there is no intercept?

Reply:
how can the `dropIntercept` handle the case with no intercept?

Reply:
It only gets called if there is an intercept.